dragonfly_doe2/doe/floor_space.py

```diff
@@ -0,0 +1,149 @@
+
+@dataclass
+class Wall:
+    """*.inp Wall object.
+    Example:
+        .. code-block::f#
+            "simple_example_dfb_Floor1_Room1 Wall_1" = EXTERIOR-WALL
+               CONSTRUCTION    = "EWall Construction"
+               LOCATION        = SPACE-V1
+               ..
+    """
+    name: str = None
+    location_index: int = None
+    wall_con_name: str = None
+
+    @classmethod
+    def from_room_seg(cls, _name: str, _bc_id: int, _wall_con_name: str):
+        indexed_id = _bc_id+1
+        return cls(str(_name+f'_Wall_{indexed_id}'), indexed_id, _wall_con_name)
+
+    def to_inp(self):
+        return f'"{self.name}" = EXTERIOR-WALL\n' \
+            f'   CONSTRUCTION   = "{self.wall_con_name}"\n' \
+            f'   LOCATION       = SPACE-V{self.location_index}\n   ..'
+
+    def __repr__(self) -> str:
+        return self.to_inp()
+
+
+@dataclass
+class Space:
+    """The Space Object.
+    Each Room2D has a Space obj. This obj contains windows, walls, doors
+    data.
+    .. code-block::f#
+        "simple_example_dfb_Floor1_Room1" = SPACE
+            SHAPE            = POLYGON
+            POLYGON          = "simple_example_dfb_Floor1_Room1 Plg"
+            C-ACTIVITY-DESC  = *Generic Office Program*
+            ..
+        "simple_example_dfb_Floor1_Room1 Wall_1" = EXTERIOR-WALL
+            CONSTRUCTION    = "EWall Construction"
+            LOCATION        = SPACE-V1
+            ..
+        "simple_example_dfb_Floor1_Room1 Wall_2" = EXTERIOR-WALL
+            CONSTRUCTION    = "EWall Construction"
+            LOCATION        = SPACE-V2
+            ..
+    """
+    name: str = None
+    activity: str = None
+    walls: List[Wall] = None
+
+    @classmethod
+    def from_room(cls, room: Room2D):
+        if not isinstance(room, Room2D):
+            # TODO: Make raise exception rather than print
+            print(
+                f'Unsupported Type: {type(room)}.\n'
+                'Expected dragonfly.room2d.Room2D'
+            )
+
+        wall_constr_name = room.properties.energy.construction_set.wall_set.exterior_construction.display_name
+        walls = []
+        bcs = [str(bc) for bc in room.boundary_conditions]
+        if 'Outdoors' in bcs:
+            for i, bc in enumerate(bcs):
+                if bc == 'Outdoors':
+                    walls.append(Wall().from_room_seg(
+                        room.display_name, i, wall_constr_name))
+        else:
+            walls = None
+
+        return cls(
+            name=room.display_name,
+            activity=room.properties.energy.program_type.display_name, walls=walls)
+
+    def to_inp(self):
+        space_walls = '\n'.join(wall.to_inp() for wall in self.walls)
+        space_block = f'"{self.name}" = SPACE\n' \
+                      f'   SHAPE           = POLYGON\n' \
+                      f'   POLYGON         = "{self.name} Plg"\n' \
+                      f'   C-ACTIVITY-DESC = *{self.activity}*\n   ..\n'
+        return space_block + space_walls
```

Comment:
I think this is easier to read and also doesn't use string concatenation:

```suggestion
                      f'   C-ACTIVITY-DESC = *{self.activity}*\n   ..'
        return '\n'.join(space_block, space_walls)
```